### PR TITLE
Converts most of our route render prop usage to children

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
https://github.com/ReactTraining/react-router/blob/dev/docs/advanced-guides/migrating-5-to-6.md#upgrade-to-react-router-v51 is the basic gist.

There are a couple of places that I didn't touch:

1) index.jsx - There are some complicated use cases in here and I didn't want to go refactoring the whole thing
2) Inventories.jsx - This component is class based.  It'll be easier to make this change when we convert it to a functional component and can use hooks.
3) Templates.jsx - see Inventories.jsx.

##### ISSUE TYPE
 - Task

##### COMPONENT NAME
 - UI

### CODE REVIEW

**Comment by mabashian** [comment]
Looks like there are some unit tests that need fixing in here

File: awx/ui_next/src/components/Breadcrumbs/Breadcrumbs.jsx [hunk 1]
```diff
@@ -7,26 +7,24 @@ import {
   BreadcrumbItem,
   BreadcrumbHeading,
 } from '@patternfly/react-core';
-import { Link, Route, withRouter } from 'react-router-dom';
+import { Link, Route, withRouter, useRouteMatch } from 'react-router-dom';
```
Comment (jlmitch5) [inline awx/ui_next/src/components/Breadcrumbs/Breadcrumbs.jsx]: I think we should probably move from using withRoute

File: awx/ui_next/src/components/Breadcrumbs/Breadcrumbs.jsx [hunk 2]
```diff
@@ -7,26 +7,24 @@ import {
...
+
 import styled from 'styled-components';
 
 const PageSection = styled(PFPageSection)`
   padding-top: 10px;
   padding-bottom: 10px;
 `;
 
-const Breadcrumbs = ({ breadcrumbConfig }) => {
+const Breadcrumbs = ({ breadcrumbConfig, match }) => {
```
Comment (jlmitch5) [inline awx/ui_next/src/components/Breadcrumbs/Breadcrumbs.jsx]: Also passing match...It'd be better to get what we need in Crumb using hooks I think

File: awx/ui_next/src/screens/Inventory/InventoryGroups/InventoryGroups.jsx [hunk 3]
```diff
@@ -11,32 +11,18 @@ import InventoryGroupsList from './InventoryGroupsList';
...
-        )}
-      />
-      <Route
-        key="list"
-        path="/inventories/inventory/:id/groups"
-        render={() => {
-          return <InventoryGroupsList />;
-        }}
-      />
+      <Route key="add" path="/inventories/inventory/:id/groups/add">
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Inventory/InventoryGroups/InventoryGroups.jsx:14]: I'm pretty sure I updated these internal inventory ones to using children as opposed to render as part of my recently merged associate/disassociate host groups pr based on review feedback.  You might want to rebase.

**Review by jlmitch5 — CHANGES_REQUESTED** [review]
just a few little things, looks awesome though, thanks for doing this!

**Review by jlmitch5 — APPROVED** [review]
looks good!

**Pull request merged**